Geometry/VeryForwardGeometryBuilder/plugins/PPSGeometryESProducer.h

```diff
@@ -0,0 +1,85 @@
+#ifndef GEOMETRY_VERYFORWARDGEOMETRYBUILDER_PPSGEOMETRYESPRODUCER_H
+#define GEOMETRY_VERYFORWARDGEOMETRYBUILDER_PPSGEOMETRYESPRODUCER_H
+
+/****************************************************************************
+*  Based on CTPPSGeometryESModule.cc by:
+*  Jan Kaspar (jan.kaspar@gmail.com)
+*  Dominik Mierzejewski <dmierzej@cern.ch>
+*
+*  Rewritten + Moved out common functionailities to DetGeomDesc(Builder) by Gabrielle Hugo.
+*  Migrated to DD4hep by Wagner Carvalho and Gabrielle Hugo.
+*
+****************************************************************************/
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/ModuleFactory.h"
+#include "FWCore/Framework/interface/ESProducer.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include "DetectorDescription/DDCMS/interface/DDCompactView.h"
+#include "CondFormats/PPSObjects/interface/CTPPSRPAlignmentCorrectionsData.h"
+#include "Geometry/VeryForwardGeometryBuilder/interface/DetGeomDesc.h"
+#include "Geometry/VeryForwardGeometryBuilder/interface/CTPPSGeometry.h"
+
+#include "CondFormats/AlignmentRecord/interface/RPRealAlignmentRecord.h"
+#include "CondFormats/AlignmentRecord/interface/RPMisalignedAlignmentRecord.h"
+#include "Geometry/Records/interface/IdealGeometryRecord.h"
+#include "Geometry/Records/interface/VeryForwardMisalignedGeometryRecord.h"
+#include "Geometry/Records/interface/VeryForwardRealGeometryRecord.h"
+
+/**
+ * \brief Builds ideal, real and misaligned geometries.
+ *
+ * First, it creates a tree of DetGeomDesc from DDCompView. For real and misaligned geometries,
+ * it applies alignment corrections (RPAlignmentCorrections) found in corresponding ...GeometryRecord.
+ *
+ * Second, it creates CTPPSGeometry from DetGeoDesc tree.
+ **/
+
+class PPSGeometryESProducer : public edm::ESProducer {
```

Comment:
For plugins it is recommended to not have a separate header, but include everything in the `.cc` file.

Reply:
thanks, fixed